azblob/zc_policy_retry.go

```diff
@@ -177,6 +177,9 @@ func NewRetryPolicyFactory(o RetryOptions) pipeline.Factory {
 
 				if !tryingPrimary {
 					requestCopy.Request.URL.Host = o.retryReadsFromSecondaryHost()
+					requestCopy.URL.Host = o.retryReadsFromSecondaryHost()
```

Comment:
If I understand correctly, by the rules of struct embedding `requestCopy.URL` is an alias for `requestCopy.Request.URL`  so I think there should be just two lines here, instead of four. Namely, there should be 
```requestCopy.Request.URL.Host = o.retryReadsFromSecondaryHost()```
and 
```requestCopy.Request.Host = o.retryReadsFromSecondaryHost()```

But not the other two lines. (Alternatively, and with the exact same meaning, ".Request." could be omitted from both the lines I suggest, to produce identical code.  I.e. out of the 4 lines currently here, there are two equivalent pairs).  Can we please get this down to one of the two-line versions?

Reply:
Ah! Good point. I was simply running through Goland's code suggestions-- probably a better idea to just look at the struct itself.